apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+# weights
+file(GLOB WF baseline.weights)
+get_filename_component(BASENAME_WF ${WF} NAME_WE)
+set(WF_CPP ${BASENAME_WF}.cpp)
+
+add_custom_command(OUTPUT ${WF_CPP}
+                   DEPENDS ${WF} binary2cpp
+                   COMMAND binary2cpp ${BASENAME_WF}_weights < ${WF} > ${WF_CPP}
+                   COMMENT "${WF} -> ${WF_CPP}"
+                   VERBATIM)
+
+# =======================================================
+
+# retrain_cost_model
+add_executable(retrain_cost_model_process
+               ASLog.cpp
+               DefaultCostModel.cpp
+               Weights.cpp
+               retrain_cost_model.cpp
+               ${WF_CPP})
+set_target_properties(retrain_cost_model_process
+                      PROPERTIES OUTPUT_NAME retrain_cost_model)
+
+set(COST_MODEL_GENERATOR_FILES cost_model_generator.cpp)
+halide_generator(cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_generator(train_cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_library_from_generator(cost_model
+                              FUNCTION_NAME
```

Comment:
Style nit: prefer ARGNAME and argvalue on the same line, which is how we do this elsewhere, e.g.

```
halide_library_from_generator(train_cost_model
                              FUNCTION_NAME train_cost_model
                              GENERATOR train_cost_model.generator)
```